keps/sig-api-machinery/5000-api-linting-crd-schema-tooling/README.md

```diff
@@ -265,20 +266,62 @@ Currently, it is proposed that:
+- The validation/validator pattern from https://github.com/everettraven/crd-diff is used. It clearly distinguishes between individual property validations and broader scoped CRD validations. Additionally, it has a validation pattern in place that checks for general CRD changes, compatibility of version-to-version, and compatibility of served versions.
 
 https://github.com/everettraven/crd-diff currently has some configuration logic in place, but it is highly repetitive and not a great user experience. This needs to be evaluated further and will likely result in a distinct configuration process for the new project.
 
-Additionally, the following validations from both projects are proposed to be included in the move to the common project:
-- **Enum** (crd-diff): Validates compatibility of changes to enum constraints on a property. Flags net new enum constraint on existing property, adding an enum, and removing an enum.
-- **Default** (crd-diff): Validates compatibility of changes to default property values. Flags adding, removing, or changing of a default value.
-- **Max\*** (crd-diff): Validates compatibility of changes to property constraints related to max value constraints (maximum, maxItems, maxLength, maxProperties). Flags adding net new max* constraints and decreasing existing max* constraints.
-- **Min\*** (crd-diff): Validates compatibility of changes to property constraints related to min value constraints (minimum, minItems, minLength, minProperties). Flags adding net new min* constraints and increasing existing min* constraints.
-- **Required** (crd-diff): Validates compatibility of required fields. Flags adding new required properties.
-- **Type** (crd-diff): Flags changes in property type.
-- **Scope** (crd-diff): Flags changes in CRD scope.
-- **ExistingFieldRemoval** (crd-diff, crd-schema-checker): Flags removal of existing fields.
-- **StoredVersionRemoval** (crd-diff): Flags removal of a stored version. Only applicable when comparing to a CRD that is already present on a cluster.
+The tables below show any existing and proposed validations for each of the projects and whether or not they will be carried over to the new common project. Any validations that behave the same in
+both projects marked as being carried over will be combined into a single validation.
+
+Validations in https://github.com/everettraven/crd-diff, both existing and proposed:
+
+| Name | Scope | Description | Implemented | Carry |
+| ---- | ----- | ----------- | ----------- | ----- |
+| enum | Property | Validates compatibility of changes to enum constraints on a property. Net new enum constraints, adding and removing enum values are flagged. | Yes | Yes |
```

Comment:
Is there an option to be able to tailor this? If we had a use case where we were confident that adding a new enum value would be sufficiently handled such that this was safe, how would we make an exception?

Reply:
There is configuration options implemented today for most of the validations here. See: https://everettraven.github.io/crd-diff/#/validations?id=enum

I didn't include configuration information in this table though, because I'm not quite happy with the existing configuration method and called this out explicitly as needing some further considerations: https://github.com/kubernetes/enhancements/pull/5102/files#diff-41e4d1a3fd8006a9e40e6415cede33325066400913bdaa49808839037eb2fecaR271

If it makes sense to add at least the configuration options that we anticipate having in some form, I can run back through and add them.

Reply:
Instead of polluting the list of validations here with all the possible configuration options, I added a link to the bottom of this section to point to the documentation site for crd-diff in https://github.com/kubernetes/enhancements/pull/5102/commits/e69b3eda1028594e6d76b513e3183e12bc8aee91